File: day1/python/day1_part1.py
def read_instruction(line: str):
    if line[0] == 'L':
        return -1 * int(line[1:])
    else:
        return int(line[1:])

def read_input(path: str):
    with open(path) as f:
        return [read_instruction(line) for line in f]

def main():
    instructions = read_input('input.txt')
    current_position = 50
    zero_count = 0
    for instruction in instructions:
        current_position += instruction
        current_position = current_position % 100
        if current_position == 0:
            zero_count += 1
    print(zero_count)

if __name__ == '__main__':
    main()

File: day1/c/day1_part1.c
#include <stdio.h>
#include <stdlib.h>

int
read_instruction(s)
char *s;
{
	if (s[0] == 'L')
		return -1 * atoi(s + 1);
	else
		return atoi(s + 1);
}

int
main()
{
	FILE *f;
	char line[256];
	int pos, cnt, instr;

	f = fopen("input.txt", "r");
	if (f == NULL) {
		perror("input.txt");
		return 1;
	}

	pos = 50;
	cnt = 0;
	while (fgets(line, sizeof(line), f) != NULL) {
		instr = read_instruction(line);
		pos = (pos + instr) % 100;
		if (pos < 0)
			pos += 100;
		if (pos == 0)
			cnt++;
	}

	fclose(f);
	printf("%d\n", cnt);
	return 0;
}
